### PR TITLE
Scheduling and uptime

### PR DESCRIPTION
**Pull request by Thelonedevil**

### CODE REVIEW

File: scripts/ScriptUptime.groovy [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    long differnce = nowDate.getTime() - startDate.getTime();
+    Date date = new Date(differnce);
+    Calendar calendar = Calendar.getInstance();
+    calendar.setTime(date);
+    int days = calendar.get(Calendar.DAY_OF_YEAR) - 1;
+    int hours = calendar.get(Calendar.HOUR) - 1;
+    int mins = calendar.get(Calendar.MINUTE);
+    int secs = calendar.get(Calendar.SECOND);
+    String response = days + (days > 1 ? "days" : "day") + ", " + hours + (hours > 1 ? "hours" : "hour") + ", " + mins + (mins > 1 ? "minutes" : "minute") + ", " + secs + (secs > 1 ? "seconds" : "second");
+    ScriptHelper.sendMessage(sArgs.destinationChannel, response, MessagePriority.HIGH)
```
Comment (NthPortal) [inline scripts/ScriptUptime.groovy]: Should be "== 1", because "0 days". Also, you probably want a space before the words/after the numbers.

Comment (Thelonedevil) [inline scripts/ScriptUptime.groovy]: >_> i changed it to that, obviously forgot to put the changes in the repo >_>

File: scripts/ScriptSchedule.groovy [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            ScriptHelper.runCommand(commandStr, sArgs.user, sArgs.channel, sArgs.destinationChannel, MessagePriority.HIGH);
+            return true;
+        default:
+            String commandStr = BuiltinCommands.GENERAL_INVALID_ARG + " " + sArgs.commandName + " " + timeUnit;
+            ScriptHelper.runCommand(commandStr, sArgs.user, sArgs.channel, sArgs.destinationChannel, MessagePriority.HIGH);
+            return false;
+    }
+
+}
+private boolean isScheduled(String channel, String command) {
```
Comment (NthPortal) [inline scripts/ScriptSchedule.groovy]: Weren't you going to add that to the API?

Comment (Thelonedevil) [inline scripts/ScriptSchedule.groovy]: I was... Thank you for reminding me.

File: scripts/ScriptUnschedule.groovy [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+public boolean execute(ScriptArgs sArgs) {
+    //!unschedule !test
+    if (sArgs.argsList.length < 1) {
+        String commandStr = BuiltinCommands.GENERAL_INSUFFICIENT_ARGS + " " + sArgs.commandName;
+        ScriptHelper.runCommand(commandStr, sArgs.user, sArgs.channel, sArgs.destinationChannel, MessagePriority.HIGH);
+        return false;
+    }
+    String command = String.join(" ", sArgs.argsList[0..-1]);
+    if (!Command.exists(sArgs.db, command)) {
```
Comment (NthPortal) [inline scripts/ScriptUnschedule.groovy]: I would leave this out, because they might delete a command and then want to unschedule it.

**Comment by NthPortal** [comment]
I just realized that there's no way to list all scheduled commands.

**Pull request merged**